drivers/flash/flash_stm32f7x.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <flash.h>
+#include <init.h>
+#include <soc.h>
+
+#include <arch/cpu.h>
+#include <cortex_m/exc.h>
+
+#include "flash_stm32.h"
+
+#define STM32F7X_SECTOR_MASK		((u32_t) 0xFFFFFC07)
+
+bool flash_stm32_valid_range(struct device *dev, off_t offset, u32_t len,
+			     bool write)
+{
+	ARG_UNUSED(write);
+
+	return flash_stm32_range_exists(dev, offset, len);
+}
+
+static int write_byte(struct device *dev, off_t offset, u8_t val)
+{
+	struct stm32f7x_flash *regs = FLASH_STM32_REGS(dev);
+	int rc;
+
+	/* if the control register is locked, do not fail silently */
+	if (regs->cr & FLASH_CR_LOCK) {
+		return -EIO;
+	}
+
+	rc = flash_stm32_wait_flash_idle(dev);
+	if (rc < 0) {
+		return rc;
+	}
+
+	/* clear PSIZE bits */
+	regs->cr &= CR_PSIZE_MASK;
+	regs->cr |= FLASH_PSIZE_BYTE;
+	regs->cr |= FLASH_CR_PG;
+
+	/* flush the register write */
+	__DSB();
+
+	*((u8_t *) offset + CONFIG_FLASH_BASE_ADDRESS) = val;
+
+	/* flush the register write */
+	__DSB();
+
+	rc = flash_stm32_wait_flash_idle(dev);
+	regs->cr &= (~FLASH_CR_PG);
+
+	return rc;
+}
+
+static int erase_sector(struct device *dev, u32_t sector)
+{
+	struct stm32f7x_flash *regs = FLASH_STM32_REGS(dev);
+	int rc;
+
+	/* if the control register is locked, do not fail silently */
+	if (regs->cr & FLASH_CR_LOCK) {
+		return -EIO;
+	}
+
+	rc = flash_stm32_wait_flash_idle(dev);
+	if (rc < 0) {
+		return rc;
+	}
+
+#if FLASH_SECTOR_TOTAL == 24
+	/* Dual bank mode, SNB MSB selects the bank2, others select sector. */
+	if (sector > 11) {
```

Comment:
By 1Mbytes devices, the sector8, sector9, sector10, sector11 are invalid. the first sector on bank2 is sector12, same as 2Mbytes devices. (table 6, RM0410)

Reply:
You are correct, I had the impression that we have to account for the missing sectors 7-11 for 1MB devices, but this will have to be done at the flash page layout level. Anyway those devices are not supported and the flash page layout API does not support defining holes in the map.